mongo/errors.go

```diff
@@ -71,6 +73,56 @@ func replaceErrors(err error) error {
 	return err
 }
 
+// IsDuplicateKeyError returns true if err is a duplicate key error
+func IsDuplicateKeyError(err error) bool {
+	if e, ok := err.(ServerError); ok {
+		return e.HasErrorCode(11000) || e.HasErrorCode(11001) || e.HasErrorCode(12582) ||
+			e.HasErrorCodeWithMessage(16460, " E11000 ")
+	}
+	return false
+}
+
+// IsTimeout returns true if err is from a timeout
+func IsTimeout(err error) bool {
+	//timeout error labels
+	if se, ok := err.(ServerError); ok {
+		if se.HasErrorLabel("NetworkTimeoutError") || se.HasErrorLabel("ExceededTimeLimitError") {
+			return true
+		}
+	}
+
+	for ; err != nil; err = unwrap(err) {
+		// check unwrappable errors together
+		if err == context.DeadlineExceeded {
+			return true
+		}
+		if ne, ok := err.(net.Error); ok {
+			return ne.Timeout()
```

Comment:
To confirm, I think this is ok.`net.Error` does not have an `Unwrap` (based on https://golang.org/pkg/net/#Error). So this cannot wrap additional errors, right?

Reply:
Ok, that seems like a reasonable assumption.